### DISCUSSION TITLE
HMR with pinia? With webpack 5?

### DISCUSSION THREAD

**Discussion by matbjo-dev** [quasarframework/quasar#12890] [General - Components / Directives / etc] [answered]
Hi, love the quasar framework. 

I am trying to get HMR to work with quasar webpack and pinia? I am trying to not have to refresh page each time code in piniastore is changed. 

End of my file workoutStore.js using quasar framework vue 3.

```javascript
if (import.meta.webpackHot) {
    import.meta.webpackHot.accept(acceptHMRUpdate(workoutStore, import.meta.webpackHot),
 (err)=>{console.log(err)}, (err)=>{console.log(err)})
}

export { workoutStore, }

```

Do you have any tips or have a github project with quasar pinia webpack and hot reload working?

Thanks.

**Comment by rstoenescu** [comment, marked as answer]
`import.meta` is a feature of Vite. But you say that you are using q/app-webpack, so it can't work. Switch to q/app-vite?
In any case, best resource is the Pinia documentation. If you don't find info about HMR with webpack there, you won't find it anywhere.
